### PR TITLE
Fixing tarball creation to contain LDAP authentication

### PR DESCRIPTION
**Pull request by intjftw**
The scripts that support tarball creation did not contain the dependencies for LDAP authentication. I complemented the scripts accordingly.

### CODE REVIEW

File: .gitlab/ci.yml [hunk 1]
```diff
@@ -116,6 +119,8 @@ tarball ubuntu-16.04:
         - apt-get install -yqq build-essential libgmp-dev zlib1g-dev
         # build tools for ctags
         - apt-get install -yqq dh-autoreconf pkg-config
+        # build tools for OpenLDAP
+        - apt install -yqq groff groff-base
```
Comment (mcserep) [inline .gitlab/ci.yml:123]: Based on OpenLDAP's CI script, only `groff-base` seems to be required:
https://git.openldap.org/openldap/openldap/-/blob/master/.gitlab-ci.yml

File: .gitlab/ci.yml [hunk 2]
```diff
@@ -95,6 +96,8 @@ tarball suse-42.1:
         # build tools for CodeCompass
         - zypper install -y -t pattern devel_basis
         - zypper install -y binutils gcc-c++ gmp-devel
+        # build tools for OpenLDAP
+        - zypper install -y groff groff-full
```
Comment (mcserep) [inline .gitlab/ci.yml:100]: Is `groff-base` not sufficient, like for Ubuntu?

File: .gitlab/ci.yml [hunk 3]
```diff
@@ -74,6 +74,7 @@ tarball suse-15:
         # build tools for CodeCompass
         - zypper install -y -t pattern devel_basis
         - zypper install -y binutils gcc-c++ gmp-devel
+        - zypper install -y groff groff-full
```
Comment (mcserep) [inline .gitlab/ci.yml:77]: Is `groff-base` not sufficient, like for Ubuntu?
(Also comment "*build tools for OpenLDAP*" missing here.)

File: .gitlab/build-deps.sh [hunk 4]
```diff
@@ -383,6 +383,23 @@ cd $DEPS_INSTALL_RUNTIME_DIR
...
+tar -xf openldap-2.5.6.tgz
+
+cd openldap-2.5.6
+./configure --quiet --prefix=$DEPS_INSTALL_RUNTIME_DIR/openldap-install
+make depend
+make
+make install
+
+rm -f $PACKAGES_DIR/openldap-2.5.6.tgz
+export PATH=$DEPS_INSTALL_RUNTIME_DIR/openldap-install/bin:$PATH
```
Comment (mcserep) [inline .gitlab/build-deps.sh:401]: Why is this required? Are the LDAP executable binaries used during the compilation?

**Review by mcserep — CHANGES_REQUESTED** [review]

File: .gitlab/build-deps.sh (hunk 4 above)
Comment (intjftw) [reply .gitlab/build-deps.sh:401]: No, mistake.

File: .gitlab/ci.yml (hunk 3 above)
Comment (intjftw) [reply .gitlab/ci.yml]: There is no supported groff-base package for SUSE 42 and 15 but groff is enough so I removed groff-full.

File: .gitlab/ci.yml (hunk 2 above)
Comment (intjftw) [reply .gitlab/ci.yml]: There is no supported groff-base package for SUSE 42 and 15 but groff is enough so I removed groff-full.

File: .gitlab/ci.yml (hunk 1 above)
Comment (intjftw) [reply .gitlab/ci.yml]: I removed groff.

**Review by mcserep — APPROVED** [review]
